io.welle.welle-gui.yml

```diff
@@ -0,0 +1,103 @@
+app-id: io.welle.welle-gui
+runtime: org.kde.Platform
+runtime-version: '6.8'
+sdk: org.kde.Sdk
+command: welle-io
+finish-args:
+  - '--share=network'
+  - '--share=ipc'
+  - '--filesystem=xdg-documents'
+  - '--socket=pulseaudio'
+  - '--socket=wayland'
+  - '--socket=fallback-x11'
+  - '--device=dri'
+  - '--device=all'
+
+modules:
+  - name: fftw
+    buildsystem: autotools
+    config-opts:
+      - '--enable-shared'
+      - '--disable-static'
+      - '--enable-threads'
+      - '--enable-float'
+    sources:
+      - type: archive
+        url: 'http://fftw.org/fftw-3.3.10.tar.gz'
+        md5: 8ccbf6a5ea78a16dbc3e1306e234cc5c
+
+  - name: libfaad
+    buildsystem: cmake
+    sources:
+      - type: git
+        url: 'https://salsa.debian.org/multimedia-team/faad2.git'
+        tag: debian/2.11.2-1
+
+  - name: libusb
+    buildsystem: autotools
+    config-opts:
+      - '--disable-udev'
+    sources:
+      - type: git
+        url: 'https://github.com/libusb/libusb.git'
+        tag: v1.0.27
+
+  - name: librtlsdr
+    buildsystem: cmake
+    config-opts:
+      - '-Wno-dev'
+      - '-DDETACH_KERNEL_DRIVER=ON'
+    sources:
+      - type: git
+        url: 'https://gitea.osmocom.org/sdr/rtl-sdr.git'
+        tag: v2.0.2
+
+  - name: libairspy
+    buildsystem: cmake
+    config-opts:
+      - '-Wno-dev'
+    sources:
+      - type: git
+        url: 'https://github.com/airspy/airspyone_host.git'
+        tag: v1.0.10
+
+  - name: SoapySDR
+    buildsystem: cmake
+    config-opts:
+      - '-DCMAKE_BUILD_TYPE=Release'
+      - '-Wno-dev'
+    sources:
+      - type: git
+        url: 'https://github.com/pothosware/SoapySDR.git'
+        tag: soapy-sdr-0.8.1
+
+  - name: welle.io
+    buildsystem: qmake
+    build-commands:
+      # Copy welle.io binary
+      - cp src/welle-gui/welle-io /app/bin
+
+      # Copy icons
+      - install -Dm644 src/welle-gui/icons/16x16/welle-io.png /app/share/icons/hicolor/16x16/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/32x32/welle-io.png /app/share/icons/hicolor/32x32/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/48x48/welle-io.png /app/share/icons/hicolor/48x48/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/128x128/welle-io.png /app/share/icons/hicolor/128x128/apps/io.welle.welle-gui.png
+      - install -Dm644 src/welle-gui/icons/256x256/welle-io.png /app/share/icons/hicolor/256x256/apps/io.welle.welle-gui.png
+      - install -Dm644 io.welle.welle-gui.desktop -t /app/share/applications
```

Comment:
these should also be installed with the upstream build system. Even qmake knows how to do it.

Reply:
Is this is a hard blocker? I mean, when I add this into the `qmake` .pro file, the files will be installed to `/app/..` for all users when they are doing `make install` in a local build. No matter if they are interested in flatpaks or not. Furthermore, this will mess up the users file system, or I'm wrong?

Do you have a reference Qt flatpak?

Reply:
`org.kde.*` but they use cmake instead.

(my opinion of qmake is even lower than cmake)

Reply:
Of course, I already checked the KDE apps. But it is a little bit hard to me to transfer the way how KDE apps are doing it to welle.io. Anyway, I will try to find a way to remove the extra install steps.

Reply:
Fixed in https://github.com/AlbrechtL/flathub/commit/fb66dbf0986effa9ed6f7e80d1cd6bede07ab510.